package.json

```diff
@@ -39,7 +39,7 @@
     "fix:prettier": "prettier \"src/**/*.js\" --write",
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "react-scripts test --env=node",
```

Comment:
Running separately, one with the node argument and one without would be good yeah, I've done that on other projects before.

Reply:
Could you give me an example of what the two test runs would be targeting generally?

Reply:
If I understand your question correctly, one would target node and would would target the browser. The `App.test.js` test fails when you pass in `--env=node` because it can't find the browser's `localStorage` variable.

So we can split up "test" into two comands: `"test:node"` and `"test:browser"`, one with and without the `--env=node` flag. The `"test"` command can then just invoke both of these.

Reply:
Thanks @armaniferrante.  I was asking about the best way to ensure the two target different files.  For instance, if the node test runs against App.test.js and possibly others in the future it gains.  On the other hand if the "not node" runs on utils.test.ts and possibly others in the future it will fail.

So I'm looking for ideas on how to generically target these tests other than specifying the individual files in the test command.  Hopefully that makes sense

Reply:
@moshthepitt Now I'm curious, why did you need the `--env=node` to begin with? All the code should be running in the browser anyway, thought about @armaniferrante 's comments for a min, and I think it actually makes sense to figure out why this didn't work without that... this is a create-react-app so there's no node outside of development.

Reply:
@evanpipta it is a work around a [known jest bug](https://github.com/facebook/jest/issues/4422) as far as `Uint8Array` is concerned.  Which means that the test I wrote will never pass because the code at [this point](https://github.com/moshthepitt/spl-token-wallet/blob/74-followup/src/utils/utils.ts#L95) complains that we do not have a valid input.  In the solana web3 repo they solve this problem [like this](https://github.com/solana-labs/solana-web3.js/blob/master/jest-environment.js).